### PR TITLE
Support taskwarrior 2.6.0 serialization format

### PR DESCRIPTION
**Pull request by imran-iq**
As discussed over email opening this PR over github.

This adds support for the new (>2.6.0) while maintaining support for the legacy (<2.5.3) (de)serialization formats for taskwarrior

### CODE REVIEW

**Comment by imran-iq** [comment]
While I have CI passing, I don't think using `serde(bound)` is the right solution as it prevents us from making `import_task` generic over version, meaning (currently) you can only import task format 2.6.0

```
pub fn import_task<Version: TaskWarriorVersion>(s: &str) -> Result<Task<Version>, Error> {
    serde_json::from_str(s).map_err(Error::from)
}
```

Results in 

```
error[E0277]: the trait bound `Version: DefaultSerDer` is not satisfied
    --> src/import.rs:25:5
     |
25   |     serde_json::from_str(s).map_err(Error::from)
     |     ^^^^^^^^^^^^^^^^^^^^ the trait `DefaultSerDer` is not implemented for `Version`
 ```

**Comment by imran-iq** [comment]
Okay got it working with a very helpful comment from here: https://github.com/serde-rs/serde/issues/1059#issuecomment-1211005392

File: src/task.rs [hunk 1]
```diff
@@ -41,11 +61,17 @@ use crate::urgency::Urgency;
 ///
 /// It is deserializeable and serializeable via serde_json, so importing and exporting taskwarrior
 /// tasks is simply serializing and deserializing objects of this type.
-#[derive(Debug, Clone, PartialEq, derive_builder::Builder)]
+///
+/// As of taskwarrior version 2.6.0 and newer, the representation of `depends` has changed from
+/// being a comma seperated string of uuid's to being a proper json array. You can select which
+/// behaviour you want at runtime by providing either [TW26] (the default) or [TW25] to `Task` as its
```
Comment (matthiasbeyer) [inline src/task.rs]:
```suggestion
/// behaviour you want at compiletime by providing either [TW26] (the default) or [TW25] to `Task` as its
```

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
Only one small thing I think was worded the wrong way... but besides that, this looks good I think!

File: src/task.rs (hunk 1 above)
Comment (imran-iq) [reply src/task.rs]: updated!

**Review by matthiasbeyer — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
awesome, thank you!

**Pull request merged**